cmd/gen-go-sample/main.go

```diff
@@ -98,34 +98,45 @@ func main() {
 }
 
 func genMethodSamples(gen *generator, sampConf schema_v1p2.SampleConfig, nofmt bool, outDir string) error {
+	
 	for _, samp := range sampConf.Samples {
-		for _, iface := range gen.gapic.Interfaces {
-			if iface.Name != samp.Service {
-				continue
-			}
-			for _, meth := range iface.Methods {
-				if meth.Name != samp.Rpc {
-					continue
-				}
-				gen.reset()
-				if err := gen.genSample(samp, meth); err != nil {
-
-					err = errors.E(err, "generating: %s:%s", iface.Name+"."+meth.Name, samp.ID)
-					log.Fatal(err)
-				}
-
-				content, err := gen.commit(!nofmt, time.Now().Year())
-				if err != nil {
-					return err
-				}
-				// Handle duplicate sample IDs
-				fname := samp.ID + ".go"
-				if err := ioutil.WriteFile(filepath.Join(outDir, fname), content, 0644); err != nil {
-					return err
+		var iface GAPICInterface
+		var method GAPICMethod
+		for _, i := range gen.gapic.Interfaces {
```

Comment:
I think you can just do
```
for _, iface = range ...
```
(no colon)
and this gets rid of the need for `i` and reassigning to `iface.

Reply:
Of course, I just realized my suggestion would require a different way of marking when a method and interface are not found. Maybe setting a simple found flag before the `break`? It's probably just fine to simply have a single error message that just says `{service}.{rpc} not found`....

Reply:
Done. We can still compare the method name and interface name with `samp.Rpc` and `samp.Service` outside the loop so we don't necessarily need a flag.